### DISCUSSION TITLE
HoloLens 2 system keyboard key detection

### DISCUSSION THREAD

**Discussion by rueilmanzambi** [microsoft/MixedRealityToolkit-Unity#10493] [Q&A] [answered]
How to detect a HoloLens2 system keyboard key such as Enter key through C# scripting in Unity-MRTK ?

**Comment by MaxWang-MS** [comment, marked as answer]
Hi @rueilmanzambi, please take a look at Unity documentation [here](https://docs.unity3d.com/ScriptReference/Input.html), and for an example you can use
> public static bool GetKeyDown([KeyCode](https://docs.unity3d.com/ScriptReference/KeyCode.html) key);

**Comment by rueilmanzambi** [comment]
@MaxWang-MS , how would we implement it in such a way that it knows that we are working with the HoloLens system keyboard? I can't see a member of TouchScreenKeyboard that would work for key detection. I don't know if there's a class that can be inherited to accomplish that task.

**Reply by MaxWang-MS** [reply]
Do you need to differentiate between HoloLens system keyboard and other types of keyboard? If not, the API I mentioned should work just fine - that API does not care whether it is HoloLens keyboard or normal physical keyboard, it should just work.

**Reply by rueilmanzambi** [reply]
> Do you need to differentiate between HoloLens system keyboard and other types of keyboard? If not, the API I mentioned should work just fine - that API does not care whether it is HoloLens keyboard or normal physical keyboard, it should just work.

Finally figured it out, thank you for the help

**Comment by talothman** [comment]
@MaxWang-MS Wondering as a follow up to this question if there's a good way to detect the `X` key being pressed on the system keyboard other than binding to the event that gets called once the keyboard gets hidden?
